### PR TITLE
`OG` add twitter card and og image

### PR DESCRIPTION
**Pull request by harshraj8843**
## Changes proposed

- Better OG images

## Why

While sharing `LinkFree` links on Discord or Twitter, I noticed that the LinkFree OG can be improved.

I have some prior experience with creating dynamic OG images. So, I created one for LinkFree 🤗 

## How

`Repo`: https://github.com/codinasion/LinkFree-OG

OG images are created by Github actions and updates every 6/12 hours (not decided yet!)

The OG images are provided through an `API` 👉🏻 `https://linkfreeog.codinasion.org`

There are two separate APIs, one for `LinkFree` and the other for `User Profile`

LinkFree: https://linkfreeog.codinasion.org
Profile: https://linkfreeog.codinasion.org/<GITHUB_USERNAME>

## Showcase

LinkFree
Profile
Another Profile

## What difference it makes (on Discord)

<p align="middle">

</p>

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/3567"></a>

### CODE REVIEW

**Comment by harshraj8843** [comment]
> **Note** There are some profile OGs missing because `LinkFree API` return `Internal Server Error`
>
> Reference 👉🏻 https://linkfree.eddiehub.io/api/users/ibrahimq21

File: pages/[username].js [hunk 1]
```diff
@@ -84,13 +84,55 @@ export default function User({ data, BASE_URL }) {
...
           property="og:url"
           content={`https://linkfree.eddiehub.io/${data.username}`}
         />
-        <meta property="og:image" content={data.avatar} />
+        <meta key="og-site-name" property="og:site_name" content="LinkFree" />
+        <meta key="og-type" property="og:type" content="website" />
+        <meta
+          key="og-image"
+          property="og:image"
+          content={`https://linkfreeog.codinasion.org/${data.username}`}
```
Comment (Panquesito7) [inline pages/[username].js:129]: Is this link even valid? 👀

Comment (harshraj8843) [reply pages/[username].js:129]: Any issue ?

👉🏻 https://linkfreeog.codinasion.org/Panquesito7

**Review by Panquesito7 — APPROVED** [review]
Looks awesome! Thanks. 🚀

**Comment by harshraj8843** [comment]
> > Note There are some profile OGs missing because `LinkFree API` return `Internal Server Error`
> > Reference 👉🏻 [linkfree.eddiehub.io/api/users/ibrahimq21](https://linkfree.eddiehub.io/api/users/ibrahimq21)

Hey @Panquesito7 

any `fix` for this 🤔

**Comment by harshraj8843** [comment]
### Update

Now able to handle all profiles 🚀 🎉 

`Link`: https://linkfreeog.codinasion.org/ibrahimq21

Fixed

### How

If LinkFree API response is `500` then `GET` profile Json directly from `Github`

**Comment by harshraj8843** [comment]
> **Note** Some profiles Json have incorrect data

### `url` vs `Url` vs `URL`

https://github.com/EddieHubCommunity/LinkFree/blob/10d7d70bccff54cf782dd656393dc18d4b9d8941/data/Drabzit.json#L8-L21

### Broken `Links`

https://github.com/EddieHubCommunity/LinkFree/blob/10d7d70bccff54cf782dd656393dc18d4b9d8941/data/Drabzit.json#L15

**Comment by harshraj8843** [comment]
Hey @eddiejaoude @Pradumnasaraf 

Could you please review this PR 👀 

Thank you in advance for your time and consideration 🤗

**Comment by eddiejaoude** [comment]
Thank you for your contribution. It is better to raise an issue first so these can be discussed without you doing work first.

It looks good, but 3rd party services like this go up and down, and therefore have a negative effect on our app.

Currently I am not keen on introducing this dependency. Maybe we can do something similar on a smaller scale as part of the application itself?

**Comment by harshraj8843** [comment]
Sure, thanks

**Pull request closed without merging**